timemachine/fe/dummy.py

```diff
@@ -64,18 +68,26 @@ def generate_dummy_group_assignments(
        latter might be desirable for efficiency but is more complicated to implement).
     """
     assert len(set(core_atoms)) == len(core_atoms)
-    assert len(list(nx.connected_components(bond_graph))) == 1
+
+    if assert_single_connected_component:
+        assert len(list(nx.connected_components(bond_graph))) == 1
```

Comment:
nit/optional: noting that this change to an optional assertion reduces the power of static checking to catch errors somewhat, because now the bond anchor is optional in the return type even if we pass `assert_single_connected_component=True` (and thus downstream code not intended to handle empty cores or disconnected graphs must assert `bond_anchor is not None`).

One way around this could be to instead define a variant `generate_dummy_group_assignments_connected` that calls this function with `assert_single_connected_component=True` and returns the old type (i.e. asserting in the types that the dummy anchor is not null). Another option could be to use `typing.overload`.

(Non-blocking, OK to clean this up later.)

Reply:
This is temporary - it will be none optional (i.e. the assertion will default to False after the refactoring is done).